### PR TITLE
Support GCP Auth

### PR DESCRIPTION
**Pull request by edwardfeng-db**
## Changes

- Support auth in GCP
  - With Google Credentials
  - With Google Service Account

## Tests

- [ ] `make test` run locally
- [ ] `make fmt` applied
- [ ] relevant integration tests applied

### CODE REVIEW

**Review by mgyucht — COMMENTED** [review]
I thought I left the review but haven't apparently. Some minor notes, but this is very close.

File: setup.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
       version=version_data['__version__'],
       packages=find_packages(exclude=["tests", "*tests.*", "*tests"]),
       python_requires=">=3.7",
-      install_requires=["requests>=2.28.1,<3"],
+      install_requires=["requests>=2.28.1,<3", "google-auth"],
```
[the remark this answers is not in the document]
Comment (mgyucht) [reply setup.py:15]: Double check this syntax, I think it means any google-auth version compatible with major version 2.

File: databricks/sdk/core.py [hunk 2]
```diff
@@ -265,6 +273,70 @@ def refreshed_headers() -> Dict[str, str]:
     return refreshed_headers
 
 
+@credentials_provider('google-credentials', ['host', 'google_credentials'])
```
[the remark this answers is not in the document]
Comment (edwardfeng-db) [reply databricks/sdk/core.py:276]: Yeah I thought so as well at the beginning but after checking more I think we do need to point it to the service account json file, which is different from the default auth json file as the service account one contains more info about the keys and secrets for signing the jwt etc. So it seems like we need this to make it work

File: databricks/sdk/core.py [hunk 3]
```diff
@@ -18,8 +19,13 @@
 from typing import (Any, BinaryIO, Callable, Dict, Iterable, Iterator, List,
                     Optional, Type, Union)
 
+import google
```
[the remark this answers is not in the document]
Comment (edwardfeng-db) [reply databricks/sdk/core.py:22]: Right, not needed, removing it

**Review by mgyucht — APPROVED** [review]
LGTM provided nightly tests pass!

File: README.md [hunk 4]
```diff
@@ -204,6 +204,31 @@ w = WorkspaceClient(host=input('Databricks Workspace URL: '),
...
+
+For example, to use Google ID authentication:
+
+```python
+from databricks.sdk import WorkspaceClient
+w = WorkspaceClient(host=input('Databricks Workspace URL: '),
+                    google_service_account=input('Google Service Account: '))
+
+```
+
```
Comment (pietern) [inline README.md:231]: @edwardfeng-db Could you take a second pass at this doc? The field names mentioned are the Go SDK field names, not the Python SDK field names (`PascalCase` vs `snake_case`).

Comment (edwardfeng-db) [reply README.md:231]: Yeah I see, sorry about this, let me update that. Some copy pasta issues

**Pull request merged**